firmware/quadruna/BMS/src/app/app_heartbeatMonitor.c

```diff
@@ -0,0 +1,53 @@
+#include "app_heartbeatMonitor.h"
+#include "app_heartbeatMonitorBoard.h"
+
+#include "app_canRx.h"
+#include "app_canTx.h"
+#include "app_canAlerts.h"
+
+HeartbeatMonitorBoard vc_hbmonitor = {
+    .getter       = app_canRx_VC_Heartbeat_get,
+    .resetter     = app_canRx_VC_Heartbeat_update,
+    .fault_setter = app_canAlerts_BMS_Warning_MissingVCHeartbeat_set,
+    .fault_getter = app_canAlerts_BMS_Warning_MissingVCHeartbeat_get,
+    .timeout_ms   = 200,
+};
+
+static bool block_faults = false;
+
+void app_heartbeatMonitor_init(bool block_faults_init)
+{
+    block_faults = block_faults_init;
+    app_heartbeatMonitorBoard_init(&vc_hbmonitor);
+}
+
+void app_heartbeatMonitor_checkIn(void)
+{
+    app_canTx_BMS_Heartbeat_set(true);
+    app_heartbeatMonitorBoard_checkIn(&vc_hbmonitor);
+}
+
+void app_heartbeatMonitor_broadcastFaults(void)
+{
+    if (block_faults)
```

Comment:
fault blocking can also be set as a flag in the `HeartbeatMonitorBoard` struct if we want the end developer to use a flag to disable faults rather than manually checking it.